sim/rogue/items_sets_pve.go

```diff
@@ -213,3 +215,110 @@ var ItemSetNightSlayerBattlearmor = core.NewItemSet(core.ItemSet{
 		},
 	},
 })
+
+var ItemSetBloodfangThrill = core.NewItemSet(core.ItemSet{
+	Name: "Bloodfang Thrill",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your opening moves have a 100% chance to make your next ability cost no energy.
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+
+			var affectedSpells []*core.Spell
+
+			aura := rogue.RegisterAura(core.Aura{
+				Label:    "Clearcasting (S03 - Item - T2 - Rogue - Damage 2P Bonus)",
+				ActionID: core.ActionID{SpellID: 467735},
+				Duration: time.Second * 15,
+				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+					affectedSpells = core.FilterSlice(
+						rogue.Spellbook,
+						func(spell *core.Spell) bool {
+							return spell != nil && spell.Cost != nil && spell.Cost.CostType() == core.CostTypeEnergy
+						},
+					)
+				},
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier -= 100 })
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					core.Each(affectedSpells, func(spell *core.Spell) { spell.Cost.Multiplier += 100 })
+				},
+				OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if aura.RemainingDuration(sim) == aura.Duration || spell.DefaultCast.Cost == 0 {
+						return
+					}
+					aura.Deactivate(sim)
+				},
+			})
+
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Damage 2P Bonus",
+				ActionID: core.ActionID{SpellID: 467708},
+				OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if !result.Landed() || !spell.ProcMask.Matches(core.ProcMaskMeleeOrRangedSpecial) {
+						return
+					}
+					switch spell {
+					case rogue.Ambush, rogue.Garrote:
+						aura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Increases main hand weapon damage by 10%
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.OnSpellRegistered(func(spell *core.Spell) {
+				if spell.ProcMask.Matches(core.ProcMaskMeleeMH) {
+					spell.DamageMultiplier *= 1.10
+				}
+			})
+		},
+			// Reduces cooldown on vanish by 2.5 minutes
+		6: func(agent core.Agent) {
+			// Implemented in Vanish.go
+		},
```

Comment:
This could be implemented here using an OnInit aura
```
rogue := agent.(RogueAgent).GetRogue()
rogue.RegisterAura(core.Aura{
  Label "S03 - Item - T2 - Rogue - Damage 6P Bonus",
  OnInit: func(aura *core.Aura, sim *core.Simulation) {
    rogue.Vanish.CD.Duration -= time.Second * 2500
  },
})
```

Reply:
completed